recipes/allegro5/all/conanfile.py

```diff
@@ -0,0 +1,313 @@
+        self.requires("minimp3/20200304")
+        self.requires("openal/1.21.1")
+        self.requires("physfs/3.0.2")
+        self.requires("opus/1.3.1")
+        self.requires("opusfile/0.12")
+        self.requires("theora/1.1.1")
+        self.requires("opengl/system")
+
+        if self.settings.os != "Windows":
+
+            self.requires("xorg/system")
+            self.requires("libalsa/1.2.5.1")
+            self.requires("pulseaudio/14.2")
+            self.requires("gtk/system")
+            
+        self.options["freetype"].with_png = False
+        self.options["freetype"].with_zlib = False
+        self.options["freetype"].with_bzip2 = False
+        self.options["freetype"].with_brotli = False
+        self.options["opusfile"].http = False
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        self.run("git clone https://github.com/liballeg/allegro5.git --depth=1 --single-branch --branch=5.2.7")
+
+    def generate(self):
+
+        zlib = self.dependencies["zlib"]
+        libpng = self.dependencies["libpng"]
+        libjpeg = self.dependencies["libjpeg"]
+        freetype = self.dependencies["freetype"]
+        libwebp = self.dependencies["libwebp"]
+        flac = self.dependencies["flac"]
+        ogg = self.dependencies["ogg"]
+        vorbis = self.dependencies["vorbis"]
+        mp3 = self.dependencies["minimp3"]
+        openal = self.dependencies["openal"]
+        physfs = self.dependencies["physfs"]
+        opus = self.dependencies["opus"]
+        opusfile = self.dependencies["opusfile"]
+        theora = self.dependencies["theora"]
+
+        if self.settings.os != "Windows":
+            alsa = self.dependencies["libalsa"]
+            pulseaudio = self.dependencies["pulseaudio"]
+
+        # Configure dependency flags for cmake
+        flags = "-Wno-dev"
+        flags += " -DSHARED=" + str(self.options.shared).lower()
+        flags += " -DWANT_DOCS=false"
+        flags += " -DWANT_DOCS_HTML=false"
+        flags += " -DWANT_EXAMPLES=false"
+        flags += " -DWANT_FONT=true"
+        flags += " -DWANT_MONOLITH=true"
+        flags += " -DWANT_TESTS=false"
+        flags += " -DWANT_DEMO=false"
+        flags += " -DWANT_RELEASE_LOGGING=false"
+        flags += " -DWANT_VORBIS=true"
+        flags += " -DWANT_MP3=true"
+
+        prefix = "lib"
+        suffix = ".a"
+        if self.settings.compiler == "Visual Studio":
+            flags += " -DWANT_STATIC_RUNTIME=" + str(self.settings.compiler.runtime == "MT").lower()
+            flags += " -DPREFER_STATIC_DEPS=true"
+            prefix = ""
+            suffix = ".lib"
+        else:
+            flags += " -DWANT_STATIC_RUNTIME=false"
+            flags += " -DPREFER_STATIC_DEPS=false"
+
+        # libpng dependency
+        tools.replace_in_file(str(os.path.join(self.build_folder, "allegro5/addons/image/CMakeLists.txt")), 
```

Comment:
I think you can use `cmake_find_package[_multi]` generator in order to create find modules needed, instead of patching these lines here and below
but if it's absolutely needed and unavoidable, I think code could be unified and extracted, below there are too many similar blocks